src/accelerate/accelerator.py

```diff
@@ -1472,6 +1473,13 @@ def prepare_model(self, model: torch.nn.Module, device_placement: bool = None, e
                 if not is_type_fsdp:
                     self.state.fsdp_plugin.set_auto_wrap_policy(model)
                     fsdp_plugin = self.state.fsdp_plugin
+
+                    # need to ensure that params are re-tied after running
+                    # param_init_fn
+                    fsdp_plugin.param_init_fn = ensure_weights_retied(
+                        fsdp_plugin.param_init_fn, model, self.device, 
+                    )
```

Comment:
Would it make more sense to have this just exist as part of the `__post_init__` in `FullyShardedDataParallelPlugin` instead? Around when we declare what `param_init_fn` is?

Reply:
right that was my initial thought to, but in that `__post_init__`. I cannot access the model. It is a similar pattern to the [call of `_fsdp_qlora_plugin_updates`](https://github.com/huggingface/transformers/blob/617b21273a349bd3a94e2b3bfb83f8089f45749b/src/transformers/trainer.py#L2226), which is done in the initial part of `Trainer._inner_training_loop`; this call internally accesses `self.model`, so its done outside of dataclass preperation